cli/preprocessor.go

```diff
@@ -32,41 +34,46 @@ func processInput(
 		fmt.Println("done")
 	}
 
-	// Feed the compiler with the input
-	if !silentFlag {
-		fmt.Print("Compiling ... ")
-	}
+	if benchmarkRuns <= 1 { // Normal compiler mode
+		// Feed the compiler with the input
+		if !silentFlag {
+			fmt.Print("Compiling ... ")
+		}
 
-	var result string
-	switch compiler {
-	case "cpp", "c++":
-		result = util.ExecuteAndWaitWithOutput("./ccomc", strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
-	case "java":
-		result = util.ExecuteAndWaitWithOutput("./ccomc-java.jar", strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
-	default:
-		log.Fatal("Invalid compiler name. Only 'cpp' and 'java' are allowed values.")
-	}
+		result := runCompilerExecutable(compiler, modeSingle, fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
 
-	if !silentFlag {
-		fmt.Println("done")
-	}
+		if !silentFlag {
+			fmt.Println("done")
+		}
 
-	// Write output
-	if outFile != "" { // To file
-		// Check if output file exists
-		if util.FileExists(outFile) && !force {
-			// Ask user if he wants to overwrite the output file
-			if !util.YesNoQuestion("The output file already exists. Do you want to overwrite it?", false) {
-				return
+		// Write output
+		if outFile != "" { // To file
+			// Check if output file exists
+			if util.FileExists(outFile) && !force {
+				// Ask user if he wants to overwrite the output file
+				if !util.YesNoQuestion("The output file already exists. Do you want to overwrite it?", false) {
+					return
+				}
 			}
+			// Write output to file
+			ioutil.WriteFile(outFile, []byte(result), 0777)
+		} else { // Print to console
+			if !silentFlag {
+				fmt.Println()
+			}
+			fmt.Println(result)
 		}
-		// Write output to file
-		ioutil.WriteFile(outFile, []byte(result), 0777)
-	} else { // Print to console
-		if !silentFlag {
-			fmt.Println()
+	} else { // Benchmarking mode
+		overallRuntime := 0
+		for i := 1; i <= benchmarkRuns; i++ {
+			start := time.Now()
+			runCompilerExecutable(compiler, modeSingle, fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
+			duration := time.Since(start).Milliseconds()
+			overallRuntime += int(duration)
+			fmt.Println(strconv.Itoa(i) + ". run: " + strconv.Itoa(int(duration)) + " ms")
```

Comment:
That moment when you need StackOverflow to figure out what the method name stands for (`itoa()`) 😂

Reply:
Yes. I also don't like such method names 😂